packages/server/pipeStreamOverResponse.ts

```diff
@@ -12,6 +12,10 @@ const pipeStreamOverResponse = (
   })
   readStream
     .on('data', (chunk: Buffer) => {
+      if (res.done) {
```

Comment:
tryEnd has been getting called after `end` has been called. this is somewhat repeatable when the rateLimiter gets hit (e.g. setAppLocation gets called too many times).

i'm not sure if this is the right spot to exit early, but a stream is asynchronous so it seems like it could be